common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -274,30 +274,43 @@ def destroy(self, request, uuid):
             )
 
         if is_refund and entitlement.is_entitlement_refundable():
-            with transaction.atomic():
-                # Revoke and refund the entitlement
-                if entitlement.enrollment_course_run is not None:
-                    self._unenroll_entitlement(
-                        entitlement=entitlement,
-                        course_run_key=entitlement.enrollment_course_run.course_id,
-                        user=request.user
+            # Revoke the Course Entitlement and issue Refund
+            log.info(
+                'Entitlement Refund requested for Course Entitlement[%s]',
+                str(entitlement.uuid)
+            )
+            refund_status = refund_entitlement(course_entitlement=entitlement)
+
+            if refund_status:
+                with transaction.atomic():
+                    entitlement.expired_at_datetime = timezone.now()
+                    entitlement.save()
+
+                    log.info(
+                        'Set expired_at to [%s] for course entitlement [%s]',
+                        entitlement.expired_at,
+                        entitlement.uuid
                     )
 
-                # Revoke the Course Entitlement and issue Refund
+                    # Revoke and refund the entitlement
+                    if entitlement.enrollment_course_run is not None:
+                        self._unenroll_entitlement(
+                            entitlement=entitlement,
+                            course_run_key=entitlement.enrollment_course_run.course_id,
+                            user=request.user
+                        )
+            else:
+                # This state is achieved in most cases by a failure in the ecommerce service to process the refund.
                 log.info(
-                    'Entitlement Refund requested for Course Entitlement[%s]',
+                    'Entitlement Refund failed for Course Entitlement [%s], alert User',
                     str(entitlement.uuid)
                 )
+                return Response(
+                    status=status.HTTP_409_CONFLICT,
```

Comment:
I decided against a 5XX error because it would indicate an error with the LMS server.  This is not a failure state in the LMS, it is a failure in another service.